manifests/exception.pp

```diff
@@ -108,23 +108,26 @@
         }
       }
       $fw_command = 'portopening'
-      validate_re($protocol,['^(TCP|UDP|ICMPv(4|6))$'])
-      if $protocol =~ /ICMPv(4|6)/ {
-        $allow_context = "protocol=${protocol}"
-      } else {
-        if empty($local_port) {
-          $local_port_cmd = ''
-        } else {
-          validate_re($local_port,['any|[0-9]{1,5}'])
-          $local_port_cmd = "${local_port_param}=${local_port}"
+      if empty($protocol) {
```

Comment:
Now that `protocol`, `local_port` and `remote_port` default to `undef` instead of empty string, I'm not sure `empty` is correct. https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/puppet/parser/functions/empty.rb#L14

Maybe something like...
```puppet
if $remote_port or $local_port {
  unless $protocol {
    fail('protocol is required, when defining local or remote port')
  }
}

if $protocol =~ /^ICMP/ {
  $allow_context = "protocol=${protocol}"
} else {
  if $local_port {
    $local_port_cmd = "${local_port_param}=${local_port}"
  } else {
    $local_port_cmd = ''
  }
  # etc...
}
```

Incomplete and untested, but hopefully you see what I mean! :)